packages/web3-eth/src/utils/prepare_transaction_for_signing.ts

```diff
@@ -36,25 +34,23 @@ import { transactionBuilder } from './transaction_builder.js';
 
 const getEthereumjsTxDataFromTransaction = (
 	transaction: FormatType<PopulatedUnsignedTransaction, typeof ETH_DATA_FORMAT>,
-) => ({
-	nonce: transaction.nonce,
-	gasPrice: transaction.gasPrice,
-	gasLimit: transaction.gasLimit ?? transaction.gas,
-	to: transaction.to,
-	value: transaction.value,
-	data: transaction.data ?? transaction.input,
-	type: transaction.type,
-	chainId: transaction.chainId,
-	accessList: (
-		transaction as FormatType<PopulatedUnsignedEip2930Transaction, typeof ETH_DATA_FORMAT>
-	).accessList,
-	maxPriorityFeePerGas: (
-		transaction as FormatType<PopulatedUnsignedEip1559Transaction, typeof ETH_DATA_FORMAT>
-	).maxPriorityFeePerGas,
-	maxFeePerGas: (
-		transaction as FormatType<PopulatedUnsignedEip1559Transaction, typeof ETH_DATA_FORMAT>
-	).maxFeePerGas,
-});
+) => {
+	const txData = { ...transaction };
```

Comment:
This function change will omit some properties entirely if not defined in the input object, while older function always includes those properties as `undefined`, potentially leading to change where its consumer was expecting presence.

Reply:
happy to add them back

Reply:
done :)